include/boost/geometry/algorithms/detail/buffer/turn_in_piece_visitor.hpp

```diff
@@ -140,7 +181,12 @@ class turn_in_piece_visitor
 
         if (piece.type == geometry::strategy::buffer::buffered_empty_side)
         {
-            return false;
+            return true;
```

Comment:
This does not make any difference - but it should have been `true`

Reply:
Does this mean that this case was not covered by a unit test? Could we add one?

Reply:
Good point. Apparently not. Those empty sides are rare. 

I'm not sure how to test it.

What happens with `false` is that partition bails out and does not verify further cases.
```
            if (! visitor.apply(**it1, **it2))
            {
                return false; // Bail out if visitor returns false
            }
```

It might be that this change fixes some issues for linestrings with flat-ends, in rare cases.

Currently I'm not available to dive deeper into it - but I can make a follow-up ticket.

Reply:
Added #1408 for this situation